### PR TITLE
Bug Fix for Qwen3 VL Moe acc issue

### PR DESCRIPTION
**Pull request by tv-karthikeya**
This PR fixes the Qwen3-VL-MoE accuracy issue, compiler issue wrt subfunction
### Main Fix:  Deepstack Accuracy Fix
  The deepstack visual merge previously used:

```
  hidden_states = hidden_states.clone()
  mixed_embeds = hidden_states + visual_embeds

  local_this = torch.where(visual_pos_masks, mixed_embeds, hidden_states)

  return local_this
```

This has been changed to:

```
  visual_mask = visual_pos_masks.to(hidden_states.dtype)
  return hidden_states + (visual_embeds * visual_mask)
```

  This is mathematically equivalent for boolean visual position masks, but avoids the torch.where merge pattern. The previous torch.where form compiled but produced poor accuracy. The mask-multiply form resolves the observed accuracy
  issue.

  ### Summary for changes related to transformer upgrade v5.5.4
  - Avoids calling the current HF **self.gate(x)** implementation inside QEfficient MoE blocks, because the newer router internally performs top-k normalization using a reduction with **.sum()**  that can export as unsupported **ReduceSum under -sub-functions**.
  - Recreates the older raw-router-logits flow explicitly with F.linear, then applies TopK and softmax over selected top-k logits.
  - Updates decode sparse MoE expert weight gathering to use pre-transposed expert weights before dynamic index_select, avoiding the problematic Gather -> Transpose pattern on dynamically selected expert tensors.

**Key Changes**
  - Router path changed from:

```
    gate_out = self.gate(x)
    router_logits, top_w, top_i = gate_out
```

 to:

```
    router_logits = F.linear(x.reshape(-1, self.gate.hidden_dim), self.gate.weight)
    top_w, top_i = torch.topk(router_logits, self.gate.top_k, dim=-1)
    top_w = F.softmax(top_w, dim=-1, dtype=torch.float)
```

  - Decode expert gather changed from:

```
    w_up = self.experts.gate_up_proj.index_select(0, idx)
    w_up = w_up.transpose(1, 2)
```

to:

`    w_up = self.experts.gate_up_proj.transpose(1, 2).index_select(0, idx)
`  

Why
  - Newer transformers Qwen3-VL-MoE router returns (router_logits, router_scores, router_indices) and performs normalization internally.
  - That normalization introduced reduction ops in the exported ONNX, which caused QAIC compile failures (segfaults).

### CODE REVIEW

**Comment by tv-karthikeya** [comment]
logs: [disagg_output.txt](https://github.com/user-attachments/files/28537011/disagg_output.txt)

**Review by vbaddi — APPROVED** [review]
